yt_dlp/extractor/opencast.py

```diff
@@ -0,0 +1,251 @@
+            'id': 'ed063cd5-72c8-46b5-a60a-569243edcea8',
+            'ext': 'mp4',
+            'title': '11 - Kryptographie - 24.11.2015',
+            'thumbnail': r're:^https?://.*\.jpg$',
+            'timestamp': 1606208400,
+            'upload_date': '20201124',
+        },
+    }
+
+    def _real_extract(self, url):
+        mobj = re.match(self._VALID_URL, url)
+        host = mobj.group('host')
+        video_id = mobj.group('id')
+
+        api_json = self._call_api(host, video_id, '', note='Downloading video JSON')
+
+        search_results = api_json.get('search-results', {})
+        if 'result' not in search_results:
+            raise ExtractorError('Video was not found')
+
+        result_dict = search_results.get('result', {})
+        if not isinstance(result_dict, dict):
+            raise ExtractorError('More than one video was unexpectedly returned.')
+
+        video = result_dict.get('mediapackage', {})
+
+        result_obj = self._parse_mediapackage(video)
+        return result_obj
+
+
+class OpencastPlaylistIE(OpencastBaseIE):
+    _VALID_URL = r'''(?x)
+                    https?://(?P<host>%s)/engage/ui/index.html\?.*?
+                    epFrom=(?P<id>%s)
+                    ''' % (
+        OpencastBaseIE._INSTANCES_RE,
+        OpencastBaseIE._UUID_RE,
+    )
+
+    _API_BASE = 'https://%s/search/episode.json?sid=%s'
+
+    _TESTS = [
+        {
+            'url': 'https://oc-video1.ruhr-uni-bochum.de/engage/ui/index.html?epFrom=cf68a4a1-36b1-4a53-a6ba-61af5705a0d0',
+            'info_dict': {
+                'id': 'cf68a4a1-36b1-4a53-a6ba-61af5705a0d0',
+                'title': 'Kryptographie - WiSe 15/16',
+            },
+            'playlist_mincount': 28,
+        },
+        {
+            'url': 'https://oc-video.ruhr-uni-bochum.de/engage/ui/index.html?e=1&p=1&epFrom=b1a54262-3684-403f-9731-8e77c3766f9a',
+            'info_dict': {
+                'id': 'b1a54262-3684-403f-9731-8e77c3766f9a',
+                'title': 'inSTUDIES-Social movements and prefigurative politics in a global perspective',
+            },
+            'playlist_mincount': 8,
+        },
+    ]
+
+    def _real_extract(self, url):
+        mobj = re.match(self._VALID_URL, url)
+        host = mobj.group('host')
+        video_id = mobj.group('id')
+
+        api_json = self._call_api(host, video_id, '', note='Downloading video JSON')
+
+        search_results = api_json.get('search-results', {})
+        if 'result' not in search_results:
+            raise ExtractorError('Playlist was not found')
+
+        result_list = search_results.get('result', {})
+        if isinstance(result_list, dict):
+            result_list = [result_list]
+
+        entries = []
+        for episode in result_list:
+            video = episode.get('mediapackage', {})
+            entries.append(self._parse_mediapackage(video))
+
+        if len(entries) == 0:
+            raise ExtractorError('Playlist has no entries')
```

Comment:
Do something like this (the formatting is off in the suggestion so make sure to run flake8 on this)
```suggestion
        entries = [
        self._parse_mediapackage(episode['mediapackage'])
        for episode in result_list 
        if episode.get('mediapackage')]
```